catkin_home/src/arm_vision/include/SegmentationNode/SegmentationNode.cpp

```diff
@@ -0,0 +1,83 @@
+#include "SegmentationNode.h"
+
+segmentation::segmentation(ros::NodeHandle nh, const std::string& camera_selection) : m_nh_(nh), camera_selection_(camera_selection) {
+    // define the subscriber and publisher
+    
+    if(camera_selection.compare("intel") == 0)
```

Comment:
Define "intel" and "kinect" as static string constants, in case they need to be used somewhere else in the future.